compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -218,15 +218,32 @@ object desugar {
    *      def f$default$2[T](x: Int) = x + "m"
    */
   private def defDef(meth: DefDef, isPrimaryConstructor: Boolean = false)(using Context): Tree =
-    addDefaultGetters(elimContextBounds(meth, isPrimaryConstructor))
+    addDefaultGetters(elimContextBounds(Nil, meth, isPrimaryConstructor))
 
-  private def elimContextBounds(meth: DefDef, isPrimaryConstructor: Boolean)(using Context): DefDef =
+  private def defDef(extParamss: List[ParamClause], meth: DefDef)(using Context): Tree =
+    addDefaultGetters(elimContextBounds(extParamss, meth, false))
+
+  private def elimContextBounds(extParamss: List[ParamClause], meth: DefDef, isPrimaryConstructor: Boolean)(using Context): DefDef =
     val DefDef(_, paramss, tpt, rhs) = meth
+
+    rhs match
+      case MacroTree(call) =>
+        cpy.DefDef(meth)(rhs = call).withMods(meth.mods | Macro | Erased)
+      case _ =>
+        cpy.DefDef(meth)(
+          name = normalizeName(meth, tpt).asTermName,
+          paramss =
+            elimContextBounds(extParamss, isPrimaryConstructor, true) ++
+            elimContextBounds(paramss, isPrimaryConstructor, false)
+        )
+  end elimContextBounds
+
+  private def elimContextBounds(paramss: List[ParamClause], isPrimaryConstructor: Boolean, ext: Boolean)(using Context): List[ParamClause] =
     val evidenceParamBuf = ListBuffer[ValDef]()
 
     def desugarContextBounds(rhs: Tree): Tree = rhs match
       case ContextBounds(tbounds, cxbounds) =>
-        val iflag = if sourceVersion.isAtLeast(`future`) then Given else Implicit
+        val iflag = if ext || sourceVersion.isAtLeast(`future`) then Given else Implicit
```

Comment:
This leads to the situation where context bound on the leading extension clause desugars to a `using` param clause but on the extension method params desugars to an `implicit` param clause, this seems inconsistent to me.
```scala
trait SemiGroup[T]:
   extension [U: Conv[T]](x: U) // using ev: Conv[T]
     def combine(y: T): T
   extension (x: T)
     def combine[U: Conv[T]](y: U): T  // implicit ev: Conv[T]
```

Reply:
My first move was to use `using` for both, but I thought keeping `implicit` wherever possible might be better for compatibility. I have to use `using` at least in the extension clause, because the "resugaring" checks for it specifically, see dotty/tools/dotc/printing/RefinedPrinter.scala line 822:
```scala
val (trailingUsing, rest3) = rest2.span(isUsingClause)
```

Reply:
I think in principle, while somewhat inconsistent, it's justifiable to use `using` in extension clauses instead of `implicit` because extensions are a new syntactic construct - there is no "migrating" Scala 2 code using them by definition, as such there are no compatibility concerns that could force desugaring to `implicit` instead of immediately to `using`.

Reply:
That was indeed my reasoning.

Reply:
Yes it does appear that the parser explicitly forbids implicit parameters in the extension clauses, so this makes sense